ospfd/ospf_abr.c

```diff
@@ -642,6 +642,14 @@ static int ospf_abr_translate_nssa(struct ospf_area *area, struct ospf_lsa *lsa)
 	old = ospf_external_info_find_lsa(area->ospf, &p);
 
 	if (old) {
+		/* Do not continue if type 5 LSA not approved */
+		if (!CHECK_FLAG(old->flags, OSPF_LSA_APPROVED)) {
+			if (IS_DEBUG_OSPF_NSSA)
+				zlog_debug(
+					"ospf_abr_translate_nssa(): LSA type 5 is not approved");
```

Comment:
in a debug situation is this enough information for the user to figure out what is going on?  What happens when there is 100's of type 5 lsa's?  How do I know which one is not right?

Reply:
@donaldsharp Okay, I just updated the debug message printing the LSA id. Now for a single LSA we will be reporting when unset, not approved and removed. Please see the logs below

```
Sep 12 11:37:02 r2 ospfd[30049]: Check for NSSA-ABR Tasks():
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_nssa_task(): Start
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_nssa_task(): unapprove translates
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_unapprove_translates(): Start
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_unapprove_translates(): approved unset on link id 5.5.5.5 <--- ***
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_unapprove_translates(): Stop
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_nssa_task(): NSSA initialize aggregates
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_nssa_task(): process translates
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_process_nssa_translates(): Start
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_process_nssa_translates(): looking at area 0.0.0.1
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_translate_nssa(): LSA Id 5.5.5.5, TRANSLATING 7 to 5
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_translate_nssa(): LSA Id 5.5.5.5 type 5 is not approved   <--- ***
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_process_nssa_translates(): Stop
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_nssa_task(): send NSSA aggregates
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_send_nssa_aggregates(): Start
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_send_nssa_aggregates(): looking at area 0.0.0.1
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_send_nssa_aggregates(): Stop
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_nssa_task(): remove unapproved translates
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_remove_unapproved_translates(): Start
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_remove_unapproved_translates(): removing unapproved translates, ID: 5.5.5.5  <--- ***
Sep 12 11:37:02 r2 ospfd[30049]: Flood/AS: NSSA TRANSLATED LSA
Sep 12 11:37:02 r2 ospfd[30049]: ospf_flood_through_interface(): DR/BDR sending upd to int eth1:192.168.0.2
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_remove_unapproved_translates(): Stop
Sep 12 11:37:02 r2 ospfd[30049]: ospf_abr_nssa_task(): Stop
```